.github/workflows/cmake.yml

```diff
@@ -94,16 +101,78 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: recursive
+      - uses: actions/checkout@v4
+        with:
+          path: apache-arrow
+          repository: apache/arrow
       - name: Prepare ccache
         run: |
           echo "CCACHE_DIR=${PWD}/ccache" >> ${GITHUB_ENV}
-      - name: Cache ccache
+      - name: Cache Groonga ccache
         uses: actions/cache@v4
         with:
           path: ccache
-          key: cmake-ubuntu-${{ matrix.cc }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**') }}
-          restore-keys: cmake-ubuntu-${{ matrix.cc }}-ccache-
+          key: cmake-ubuntu-groonga-${{ matrix.cc }}-ccache-${{ hashFiles('lib/**', 'src/**', 'plugins/**', 'include/**') }}
+          restore-keys: |
+            cmake-ubuntu-groonga-${{ matrix.cc }}-ccache-
+            cmake-ubuntu-groonga-ccache-
+      - name: Configure ccache for Apache Arrow
+        if: env.APACHE_ARROW == 'main'
+        run: |
+          echo "CCACHE_DIR_ARROW=${PWD}/ccache-arrow" >> ${GITHUB_ENV}
+          echo "CMAKE_CXX_COMPILER_LAUNCHER_ARROW=ccache" >> ${GITHUB_ENV}
+      - name: Cache Apache Arrow ccache
+        if: env.APACHE_ARROW == 'main'
+        uses: actions/cache@v4
+        with:
+          path: ccache-arrow
+          key: cmake-ubuntu-apache-arrow-${{ matrix.cc }}-ccache-${{ hashFiles('apache-arrow/cpp/src/**') }}
+          restore-keys: |
+            cmake-ubuntu-apache-arrow-${{ matrix.cc }}-ccache-
+            cmake-ubuntu-apache-arrow-ccache-
+      - name: Install packages
+        run: |
+          sudo apt update -o="APT::Acquire::Retries=3"
+          sudo apt install -y -V -o="APT::Acquire::Retries=3" \
+            build-essential \
+            ninja-build \
+            ccache \
+            cmake
+      - name: Configure Apache Arrow
+        if: env.APACHE_ARROW == 'main'
+        env:
+          CCACHE_DIR: ${{ env.CCACHE_DIR_ARROW }}
+        run: |
+          cd apache-arrow
+          ccache --show-stats --verbose --version || :
+          cmake \
+            -B cpp.build \
+            -S cpp \
+            -GNinja \
+            --preset ninja-release \
+            -DCMAKE_INSTALL_PREFIX=$PWD/install \
```

Comment:
Can we use the same install directory as Groonga's one to simplify our CI configuration?

Reply:
fix: [a6a338e](https://github.com/groonga/groonga/pull/2459/commits/a6a338e1dacfe3209489a82c2905d92489acfb75) Thanks, yes we can.